admin-portal/src/App.tsx

```diff
@@ -46,11 +51,15 @@ function App() {
     SupplierService.http = http;
   }, []);
 
-  if (isSigningIn) {
-    return <p>Loading...</p>;
-  }
+  useEffect(() => {
+    if(state.isAuthenticated){
+      setLoading(false);
+    }
+  });
+  
   return (
     <div className="App">
+      <LoadingOverlay active={loading} spinner={<HashLoader color={'#09d3ac'} loading={loading} cssOverride={override} size={50} />}   >
       <header className="App-header">
```

Comment:
We should avoid rendering other components until the sign-in is completed. Otherwise, those components will try to fetch data from APIs and eventually fail to render content because of authentication errors.

Reply:
done some changes, Please check and let me know